# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/localstorage/ActorsParent.cpp
# Commit: 6180f2e753bd
# Full Hash: 6180f2e753bd2efb4905c6a70eb6607144807d5b
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-11 10:49:37
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add support for initializing corresponding origin to QuotaManager::OpenClientDirectory; r=devtools-reviewers,nchevobbe
#   
#   QuotaManager::OpenClientDirectory can now get initialization status of
#   corresponding origin and initialize the origin if it's needed.
#   
# ==============================================================================

diff -r f25dfc157ef7 -r 6180f2e753bd dom/localstorage/ActorsParent.cpp
--- a/dom/localstorage/ActorsParent.cpp	Fri Oct 11 00:06:08 2024 +0000
+++ b/dom/localstorage/ActorsParent.cpp	Fri Oct 11 00:06:09 2024 +0000
@@ -6799,6 +6799,7 @@
 
   quotaManager
       ->OpenClientDirectory({mOriginMetadata, mozilla::dom::quota::Client::LS},
+                            /* aCreateIfNonExistent */ false,
                             SomeRef(mPendingDirectoryLock))
       ->Then(
           GetCurrentSerialEventTarget(), __func__,